scenarios/data_access/client1/init_site.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.asymmetric.rsa import (
+        generate_private_key, RSAPrivateKey, RSAPublicKey)
+from cryptography.hazmat.primitives.serialization import (
+    Encoding, PublicFormat)
+import requests
+
+from mahiru.definitions.assets import ComputeAsset, ComputeMetadata, DataAsset
+from mahiru.definitions.registry import PartyDescription, SiteDescription
+from mahiru.policy.rules import (
+        MayAccess, ResultOfComputeIn, ResultOfDataIn)
+from mahiru.rest.registry_client import RegistrationRestClient
+from mahiru.rest.internal_client import InternalSiteRestClient
+
+
+ASSET_DIR = Path.home() / 'mahiru' / 'assets'
+
+
+def register(public_key: RSAPublicKey) -> None:
+    """Registers the current party and site with the registry."""
+    party_id = 'party:party1_ns:party1'
+    party_desc = PartyDescription(party_id, public_key)
+
+    site_id = 'site:party1_ns:site1'
+    endpoint = 'http://site1'
+    namespace = 'party1_ns'
+    site_desc = SiteDescription(
+            site_id, party_id, party_id, endpoint, True, True, namespace)
+
+    client = RegistrationRestClient("http://registry")
+
+    # Remove stale registrations, if any
+    try:
+        client.deregister_site(site_id)
+    except KeyError:
+        pass
+    try:
+        client.deregister_party(party_id)
+    except KeyError:
+        pass
+
+    client.register_party(party_desc)
+    client.register_site(site_desc)
+
+
+def add_initial_assets(client: InternalSiteRestClient) -> None:
+    output_base = DataAsset(
+            'asset:party1_ns:da.data.output_base:party1_ns:site1', None,
+            f'{ASSET_DIR}/data-asset-base.tar.gz')
+
+    script_metadata = ComputeMetadata({
+            'output0': 'asset:party1_ns:da.data.output_base:party1_ns:site1'})
+
+    script = ComputeAsset(
+            'asset:party1_ns:da.software.script1:party1_ns:site1', None,
+            f'{ASSET_DIR}/compute-asset.tar.gz',
+            script_metadata)
+
+    assets = [output_base, script]
+
+    for asset in assets:
+        client.store_asset(asset)
+
+
+def add_initial_rules(
```

Comment:
Maybe you can add some documentation here. Maybe in the form of small natural language comments like "site1 may access it's own output asset". Because these rules basically define the behaviour of the system right? Also might be nice to add a short description of the bigger picture here, I quite like the one in the description of this PR.